kitsune/search/v2/base.py

```diff
@@ -0,0 +1,48 @@
+from django.utils import timezone
+
+from elasticsearch_dsl import Document as DSLDocument
+from elasticsearch_dsl import field
+
+
+class SumoDocument(DSLDocument):
+    """Base class with common methods for all the different documents."""
+
+    indexed_on = field.Date()
+
+    @classmethod
+    def prepare(cls, instance):
+        """Prepare an object given a model instance"""
+        obj = cls()
+        doc_mapping = obj._doc_type.mapping
+        fields = [f for f in doc_mapping]
+        fields.remove("indexed_on")
+
+        # Loop through the fields of each object and set the right values
+        for f in fields:
+
+            try:
+                # This will allow child classes to have own methods in the form of prepare_field
+                prepare_method = getattr(obj, "prepare_{}".format(f))
+                value = prepare_method(instance)
+            except AttributeError:
+                value = getattr(instance, f)
+            # Assign values to each field. Assign a dictionary to Objects
+            if isinstance(doc_mapping.resolve_field(f), field.Object):
+                locale = cls.prepare_locale(instance)
+                if locale and value:
+                    obj[f] = {locale: value}
+                # We  populate non-locale object fields here
+            else:
+                setattr(obj, f, value)
+
+        obj.meta.id = instance.id
+        obj.indexed_on = timezone.now()
+
+        return obj
+
+    @classmethod
+    def prepare_locale(cls, instance):
+        """Return the locale of an object if exists."""
+        if getattr(instance, "locale"):
+            return instance.locale
+        return ""
```

Comment:
This being a classmethod feels funny to me when all the other prepare_methods are instance methods. I think it should possibly also just return None, or even raise NotImplementedError, for subclasses to implement themselves. Again this might become clearer when we have a number of Document classes.

Reply:
I am removing the `classmethod` for now. Let's see how it's going to behave with the rest of the docs and adapt if needed.